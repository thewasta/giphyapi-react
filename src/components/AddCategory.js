import React, {useState} from "react";
import PropTypes from "prop-types";

export const AddCategory = ({setCategories}) => {
    const [inputValue, setInputValue] = useState("");
    const handleOnChange = (e) => {
        setInputValue(e.target.value);
    };
    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 1)
            setCategories(categories => [inputValue, ...categories]);
    };
    return (

        <form onSubmit={handleSubmit}>
            <input
                type="text"
                value={inputValue}
                onChange={handleOnChange}
            />
        </form>

    );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};