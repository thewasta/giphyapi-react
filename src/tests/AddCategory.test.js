import React from "react";
import {shallow} from "enzyme";
import {AddCategory} from "../components/AddCategory";

describe("Pruebas en el componente <AddCategory/>", () => {

    const setCategories = jest.fn();
    let wrapper;
    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories}/>);
    });
    test("Comprobar el snapshot", function (assert) {
        expect(wrapper).toMatchSnapshot();
        assert();
    });
    test("Debe cambiar la caja de test", function (assert) {
        const input = wrapper.find("input");
        const value = "Hola mundo";
        input.simulate("change", {target: {value}});
        expect(wrapper.find("p").text().trim()).toBe(value);
        assert();
    });
    test("NO debe de trigger submit", function (assert) {
        wrapper.find("form").simulate("submit", {
            preventDefault() {
            }
        });
        expect(setCategories).not.toHaveBeenCalled();
        assert();
    });
    test("debe de llamar setCategories y limpiar caja de texto", function (assert) {
        const value = "Nuevo valor";
        wrapper.find("input").simulate("change", {target: {value}});
        wrapper.find("form").simulate("submit", {
            preventDefault() {
            }
        });
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).not.toHaveBeenCalledTimes(2);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function)); // Comprobar de que setCategories se ha llamado como una funci√≥n
        expect(wrapper.find("input").prop("value")).toBe("");
        assert();
    });
});
