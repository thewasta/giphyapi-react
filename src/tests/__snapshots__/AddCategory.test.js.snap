// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <AddCategory/> Comprobar el snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
